File: CS/SimpleLink.cs
using System;
using System.Drawing;
using DevExpress.XtraPrinting;

namespace PrintCustomPaperSize
{
    class SimpleLink : Link
    {
        private string text;

        public string Text
        {
            get { return text; }
            set { text = value; }
        }

        public SimpleLink()
        {

        }

        public SimpleLink(PrintingSystem ps, string text)
            : base(ps)
        {
            this.text = text;
        }

        protected override void CreateDetail(BrickGraphics graph)
        {
            base.CreateDetail(graph);

            SizeF textSize = graph.MeasureString(Text, (int)Math.Round(graph.ClientPageSize.Width));

            RectangleF rect = new RectangleF(0, 0, graph.ClientPageSize.Width, textSize.Height);

            graph.DrawString(text, rect);
        }
    }
}

File: VB/SimpleLink.vb
Imports System
Imports System.Drawing
Imports DevExpress.XtraPrinting

Namespace PrintCustomPaperSize

    Friend Class SimpleLink
        Inherits Link

        Private textField As String

        Public Property Text As String
            Get
                Return textField
            End Get

            Set(ByVal value As String)
                textField = value
            End Set
        End Property

        Public Sub New()
        End Sub

        Public Sub New(ByVal ps As PrintingSystem, ByVal text As String)
            MyBase.New(ps)
            textField = text
        End Sub

        Protected Overrides Sub CreateDetail(ByVal graph As BrickGraphics)
            MyBase.CreateDetail(graph)
            Dim textSize As SizeF = graph.MeasureString(Text, CInt(Math.Round(graph.ClientPageSize.Width)))
            Dim rect As RectangleF = New RectangleF(0, 0, graph.ClientPageSize.Width, textSize.Height)
            graph.DrawString(textField, rect)
        End Sub
    End Class
End Namespace
